lab-trent/test/greet-test.js

```diff
@@ -0,0 +1,16 @@
+const assert = require('assert');
+const greet = require('../greet');
+
+describe('Greet', function() {
+  describe('greet()', function() {
+    it('should greet the user by the username passed in as a parameter', function() {
+      assert.equal(greet.greet('tom'), 'hello tom');
+    });
+    it('should greet the user by the different username passed in as a parameter', function() {
+      assert.equal(greet.greet('richard'), 'hello richard');
+    });
+    it('should greet the user by the username passed as a process argument', function() {
+      assert.equal(greet.greet(), 'hello ' + process.argv[2]);
```

Comment:
You can mock `process.argv` by actually assigning values to it inside of this scoped test, rather than trying to manage the 'getter' at this point.